### PR TITLE
Add RKE2 boostrap provider to e2e

### PR DESCRIPTION
**Pull request by alexander-demicev**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:

**Special notes for your reviewer**:

**Checklist**:

- [ ] squashed commits into logical changes
- [ ] includes documentation
- [ ] adds unit tests
- [ ] adds or updates e2e tests

### CODE REVIEW

**Review by salasberryfin — DISMISSED** [review]

**Review by richardcase — COMMENTED** [review]
Looks good to me. 

Do we need to add a custom LB configuration to handle the join port?

File: test/e2e/data/capi-operator/full-providers.yaml [hunk 1]
```diff
@@ -25,4 +35,24 @@ metadata:
...
+---
+apiVersion: operator.cluster.x-k8s.io/v1alpha1
+kind: BootstrapProvider
+metadata:
+  name: bootstrap-caprke2
+  namespace: rke2-bootstrap-system
+spec:
+  fetchConfig:
+    url: https://github.com/rancher-sandbox/cluster-api-provider-rke2/releases/v0.1.1/bootstrap-components.yaml
+  version: v0.1.1
```
Comment (richardcase) [inline test/e2e/data/capi-operator/full-providers.yaml:48]: We can update to use `v0.2.2` now.

**Review by richardcase — COMMENTED** [review]
Seems good to me. 

@alexander-demicev - when you get some time could you rebase this?

**Comment by salasberryfin** [comment]
Looks like this needs a rebase.

**Comment by furkatgofurov7** [comment]
@alexander-demicev beta release of CAPI is out https://github.com/kubernetes-sigs/cluster-api/releases/tag/v1.7.0-beta.0, let's rebase this and get going?

**Comment by alexander-demicev** [comment]
closing in favor of https://github.com/rancher/turtles/pull/177

**Pull request closed without merging**